aws/session/session.go

```diff
@@ -404,6 +404,10 @@ func (s *Session) clientConfigWithErr(serviceName string, cfgs ...*aws.Config) (
 			func(opt *endpoints.Options) {
 				opt.DisableSSL = aws.BoolValue(s.Config.DisableSSL)
 				opt.UseDualStack = aws.BoolValue(s.Config.UseDualStack)
+
+				// Suppor the condition where the service is modeled but its
+				// endpoint metadata is not available.
+				opt.ResolveUnknownService = true
```

Comment:
We are defaulting this to true?

I wonder if we should just always resolve it and not have a boolean. I understand that adding it will allow users to disable if needed, but I just can't think of any meaningful use cases for this.

Thoughts?